File: 8vim/src/main/java/inc/flide/vim8/keyboardhelpers/InputMethodViewHelper.java
package inc.flide.vim8.keyboardhelpers;

import static inc.flide.vim8.AppPrefsKt.appPreferenceModel;

import android.content.res.Configuration;
import android.content.res.Resources;
import android.util.DisplayMetrics;
import inc.flide.vim8.AppPrefs;
import inc.flide.vim8.R;
import inc.flide.vim8.geometry.Dimension;

public final class InputMethodViewHelper {
    private InputMethodViewHelper() {
    }

    public static Dimension computeDimension(Resources resources) {
        AppPrefs prefs = appPreferenceModel().java();
        DisplayMetrics displayMetrics = resources.getDisplayMetrics();

        float minBaseSize;
        if (resources.getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE) {
            minBaseSize = resources.getFraction(R.fraction.inputView_minHeightFraction, displayMetrics.heightPixels,
                    displayMetrics.heightPixels);
        } else {
            minBaseSize = resources.getFraction(R.fraction.inputView_minHeightFraction, displayMetrics.widthPixels,
                    displayMetrics.widthPixels);
        }
        float maxBaseSize = resources.getFraction(R.fraction.inputView_maxHeightFraction, displayMetrics.heightPixels,
                displayMetrics.heightPixels);
        int height = (int) (Math.max((minBaseSize + maxBaseSize) / 2.0f,
                resources.getDimension(R.dimen.inputView_baseHeight)));
        int scale = prefs.getKeyboard().getHeight().get();
        height = height * scale / 100;
        return new Dimension(displayMetrics.widthPixels, height);
    }

}

File: 8vim/src/main/java/inc/flide/vim8/keyboardHelpers/InputMethodViewHelper.kt
package inc.flide.vim8.keyboardHelpers

import android.content.res.Configuration
import inc.flide.vim8.geometry.Dimension
import kotlin.math.roundToInt

object InputMethodViewHelper {
    fun onMeasureHelper(width: Int, height: Int, orientation: Int): Dimension {
        var width = width
        var height = height
        if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
            //Landscape is just un-usable right now.
            // TODO: Landscape mode requires more clarity, what exactly do you want to do?
            width = (1.33f * height).roundToInt()
        } else { // Portrait mode
            height = (0.8f * (width - 60 * 3)).roundToInt()
        }
        return Dimension(width, height)
    }
}